ecl/hqlcpp/hqlhtcpp.cpp

```diff
@@ -18711,22 +18763,35 @@ void HqlCppTranslator::buildConnectOrders(BuildCtx & ctx, ABoundActivity * slave
 }
 
 
-ColumnToOffsetMap * HqlCppTranslator::queryRecordOffsetMap(IHqlExpression * record)
+bool HqlCppTranslator::useRowAccessorClass(IHqlExpression * record, bool isTargetRow)
+{
+    if (isTargetRow)
+        return false;
+    if (isFixedRecordSize(record))
+        return false;
+    if (!canCreateRtlTypeInfo(record))
+        return false;
+    return getVarSizeFieldCount(record, true) >= options.varFieldAccessorThreshold;
```

Comment:
It is often still worth using it for that case - since if you are iterating through a child dataset, calculating the size of the record will use that last offset.  This is a heuristic to prevent too much overhead on simple cases I don't think it needs to be too accurate so I would leave as it is.